### ISSUE TITLE
Allow mixed case in from base functions

### ISSUE THREAD

**Issue by dain** [facebook/presto#2669]
For example:
```
com.google.common.io.BaseEncoding$DecodingException: Unrecognized character: a
java.lang.IllegalArgumentException
com.google.common.io.BaseEncoding.decode(BaseEncoding.java:228)
com.facebook.presto.operator.scalar.VarbinaryFunctions.fromHexVarchar(VarbinaryFunctions.java:98)
```

Also invalid input should be classified as `INVALID_FUNCTION_ARGUMENT`

**Comment by haozhun** [comment]
@dain 

https://github.com/google/guava/issues/1886

A workaround as suggested by the above post is call uppercase on the input. (Or a custom simplified custom uppercase that works on slice and handles only a-z. `StringFunctions.upper` seems to be the exact function I want here if it will never support unicode.)

**Comment by haozhun** [comment]
Fixed by #2921
